### PR TITLE
Login with custom callback and token request in client (SDK)

### PR DESCRIPTION
**Pull request by pablogamium**
## Summary

Rework to work with a client callback page:
- Reworked popup mode:
  - Removed post messages
  - Added popup mode with redirect uri
- Added redirect mode
- Added email auth mode prop to select `popup` or `redirect`
- `DiditEmailAuth` provider refactor
- Removed email auth service props
- Updated documentation with all changes

## Issues

JIRA tickets:

- No ticket

Depends on:
- https://github.com/didit-protocol/didit-sdk/pull/44
- https://github.com/didit-protocol/service-didit-auth/pull/153

## Demos

Demo of popup and redirect auth modes:

https://github.com/didit-protocol/didit-sdk/assets/106668657/5e205417-958c-4a96-93c4-a021b5185177
https://github.com/didit-protocol/didit-sdk/assets/106668657/f77b96a2-0f48-4876-aa90-289185c0cdf7

### CODE REVIEW

File: packages/didit-sdk/src/contexts/diditEmailAuthContext/DiditEmailAuthProvider.tsx [hunk 1]
```diff
@@ -207,24 +254,51 @@ const DiditEmailAuthProvider = ({
   );
 
   const loginWithEmail = useCallback(
-    () => console.warn("NotImplementedError: Email login isn't supported yet"),
+    () => console.warn("NotImplementedError: Email login isn't supzported yet"),
```
Comment (pablogamium) [inline packages/didit-sdk/src/contexts/diditEmailAuthContext/DiditEmailAuthProvider.tsx:257]:
```suggestion
    () => console.warn("NotImplementedError: Email login isn't supported yet"),
```

File: README.md [hunk 2]
```diff
@@ -89,39 +90,41 @@ import { DiditAuthProvider} from 'didit-sdk';
...
+import { DiditAuthProvider, DiditAuthMethod,  } from 'didit-sdk';
 
 ...
       <DiditAuthProvider
         authMethods={[DiditAuthMethod.APPLE, DiditAuthMethod.GOOGLE]}
-        emailAuthBaseUrl="https://my.didit.app/auth/email"
         walletAuthBaseUrl="https://my.didit.app/auth/wallet"
         clientId="676573"
-        claims="read:email write:email read:profile read:blockchain"
+        claims="read:emails write:emails read:profile read:blockchain"
```
Comment (arosasg) [inline README.md:125]: read:emails write:emails is not for external applications, so I would only add read:email and write:email

Also, read:profile does not exist... so please remove

File: examples/with-nextjs/.env.example [hunk 3]
```diff
@@ -12,11 +12,8 @@ NEXT_PUBLIC_APP_NAME=Didit SDK demo
 NEXT_PUBLIC_WALLET_CONNECT_PROJECT_ID=b0337f8e2c56c722a1fb3a4cdf893249
 
 # for authentication to didit auth
-NEXT_PUBLIC_DIDIT_EMAIL_AUTH_BASE_URL=https://apx.dev.didit.me/auth
 NEXT_PUBLIC_DIDIT_WALLET_AUTH_BASE_URL=./api
 NEXT_PUBLIC_DIDIT_CLIENT_ID=676573
-NEXT_PUBLIC_DIDIT_CLAIMS="read:email write:email read:profile read:blockchain"
+NEXT_PUBLIC_DIDIT_CLAIMS="read:emails write:emails"
```
Comment (arosasg) [inline examples/with-nextjs/.env.example:17]: read:email write:email

**Review by rosasalberto — COMMENTED** [review]
Looks good

**Review by Keddib — APPROVED** [review]

**Pull request merged**